apps/web/src/hooks/loadables/useLoadBalances.ts

```diff
@@ -24,46 +24,56 @@ export const useTokenListSetting = (): boolean | undefined => {
   return isTrustedTokenList
 }
 
-export const useLoadBalances = (): AsyncResult<SafeBalanceResponse> => {
-  const [pollCount, resetPolling] = useIntervalCounter(POLLING_INTERVAL)
-  const currency = useAppSelector(selectCurrency)
-  const isTrustedTokenList = useTokenListSetting()
-  const { safe, safeAddress } = useSafeInfo()
+const useCounterfactualBalances = (safe: ExtendedSafeInfo) => {
   const web3 = useWeb3()
   const chain = useCurrentChain()
-  const chainId = safe.chainId
+  const safeAddress = safe.address.value
+  const isCounterfactual = !safe.deployed
+
+  const [data] = useAsync<SafeBalanceResponse | undefined>(() => {
+    if (!chain || !web3 || !isCounterfactual) return
+    return getCounterfactualBalance(safeAddress, web3, chain)
+  }, [chain, safeAddress, web3, isCounterfactual])
 
-  // Re-fetch assets when the entire SafeInfo updates
-  const [data, error, loading] = useAsync<SafeBalanceResponse | undefined>(
-    () => {
-      if (!chainId || !safeAddress || isTrustedTokenList === undefined) return
+  return data as unknown as Balances
+}
 
-      if (!safe.deployed) {
-        return getCounterfactualBalance(safeAddress, web3, chain)
-      }
+export const useRtkBalances = () => {
+  const currency = useAppSelector(selectCurrency)
+  const isTrustedTokenList = useTokenListSetting()
+  const { safe, safeAddress, safeLoaded } = useSafeInfo()
+  const isReady = safeLoaded && isTrustedTokenList !== undefined
 
-      return getBalances(chainId, safeAddress, currency, {
-        trusted: isTrustedTokenList,
-      })
+  const { data, isLoading, error } = useBalancesGetBalancesV1Query(
+    {
+      chainId: safe.chainId,
+      safeAddress,
+      fiatCode: currency,
+      trusted: isTrustedTokenList,
+    },
+    {
+      skip: !isReady,
```

Comment:
We also need to skip / prevent this request in case it is a counterfactual safe otherwise we are getting a 404 and an unhandled error
<img width="594" alt="Screenshot 2025-01-27 at 16 27 57" src="https://github.com/user-attachments/assets/89940e23-9c0e-4acd-b07b-ee7ab5171955" />